platform independent script to verify sha256 checksums

KASR opened the pull request:
feat: add python script to verify the file hashes, the goal is to be platform independent

the output is a table with the filename, valid checksum (if the hash matches), and file missing (if the file is not found).

KASR committed:
python script to verify the checksum of the llama models

Added Python script for verifying SHA256 checksums of files in a directory, which can run on multiple platforms. Improved the formatting of the output results for better readability.

KASR committed:
Update README.md

update to the readme for improved readability and to explain the usage of the python checksum verification script

prusnak commented on scripts/verify-checksum-models.py:
```diff
@@ -0,0 +1,64 @@
+    if os.path.exists(file_path):
+        # Calculate the SHA256 checksum of the file using hashlib
+        with open(file_path, "rb") as f:
+            file_hash = hashlib.sha256(f.read()).hexdigest()
```
`f.read()` reads the whole file into the memory - this will behave poorly for big files that do not fit in RAM

you want to do something akin to:

``` python
def sha256sum(filename):
    BLOCKSIZE = 16 * 1024 * 1024
    h  = hashlib.sha256()
    b  = bytearray(BLOCKSIZE)
    mv = memoryview(b)
    with open(filename, 'rb', buffering=0) as f:
        n = f.readinto(mv)
        if n < BLOCKSIZE:
            h.update(mv[:n])
            break
        else:
            h.update(mv)
    return h.hexdigest()
```

KASR replied:
Thank you for the suggestion, I've updated the script.

KASR committed:
update the verification script

I've extended the script based on suggestions by @prusnak

The script now checks the available RAM, is there is enough to check the file at once it will do so. If not the file is read in chunks.

KASR committed:
minor improvment

small change so that the available ram is checked and not the total ram

KASR committed:
remove the part of the code that reads the file at once if enough ram is available

based on suggestions from @prusnak i removed the part of the code that checks whether the user had enough ram to read the entire model at once. the file is now always read in chunks.

KASR committed:
Merge branch 'ggerganov:master' into master

KASR committed:
Update verify-checksum-models.py

quick fix to pass the git check
